<a href="https://colab.research.google.com/github/mohdkhairilabdulkarim/STQD6014-Data-Science/blob/main/Week02_STQD6014.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Week02: Python Basic**

#Mohd Khairil Abdul Karim
##Mohd Khairil Abdul Karim
###Mohd Khairil Abdul Karim
#**Mohd Khairil Abdul Karim**

In [1]:
3+5

8

In [2]:
# Print the message
#single quote
print('hello world!')

hello world!


In [3]:
#double quote
print("hello world!")

hello world!


In [7]:
# Range of 15
x=range(15)
print(x)

# For loop (to examine each element in variable)
# mind the syntax -> space of four for subsequent line
for value in x:
  print(value**2)

range(0, 15)
0
1
4
9
16
25
36
49
64
81
100
121
144
169
196


In [8]:
# For loop (to examine each element in variable)
# mind the syntax -> space of four for subsequent line
for i in x:
  print(i**2)

0
1
4
9
16
25
36
49
64
81
100
121
144
169
196


# **Exercise 1: Multiplication**
Suppose you have RM100, which you can invest with a 10% return each year. After one year, it's 100×1.1=110 dollars, and after two years it's 100×1.1×1.1=121.

* Write a code to calculate how much money you end up with after 7 years
* print the result.

In [12]:
# How much is our $100 worth after 7 years?
# x=100(1.1)^7
x=100*1.1**7
print(x)

194.87171000000012


# **BMI calculation**

In [15]:
# Information of height and weight
height=1.75
#weight=67.3
weight=74.7

# BMI calculation
BMI=weight / height**2

# BMI value
print(BMI)

# Don't use print
BMI

# What is the new BMI if now the weight is 74.7?


24.39183673469388


24.39183673469388

# **Different Python type**

In [16]:
# float type
type(BMI)

float

In [17]:
# integer type
day_of_week=7
type(day_of_week)

int

In [19]:
# string type
x="body mass index"
y="stqd6014"
type(x)
type(y)

print(type(x))
print(type(y))

<class 'str'>
<class 'str'>


In [20]:
# string type
x=89
y="stqd6014"
type(x)
type(y)
#this one output both data type
print(type(x))
print(type(y))

<class 'int'>
<class 'str'>


In [23]:
# boolean type
print(type(True))
print(type(False))

<class 'bool'>
<class 'bool'>


# **Different Python Data type = Different Behaviour**

In [24]:
# Summation of numbers
2+3

5

In [25]:
# Summation of strings
'ab'+'cde'

'abcde'

# **Exercise 2: Python as calculator**

In [26]:
# Create a variable savings
savings=100

# Create a variable growth_multiplier
growth_multiplier=1.1

# Calculate result
result=savings*growth_multiplier**7

# Print out result
result

194.87171000000012

# **Exercise 3: String and boolean**

In [27]:
# Create a variable desc
desc="compound interest"

# Data type for desc
type(desc)

str

In [28]:
# Create a variable profitable
profitable=True

# Data type of profitable
type(profitable)

bool

# **Exercise 4: Python data type**

In [30]:
# Given the following data
a = 194.87171000000012
b = 'true'
c = False

In [31]:
# Data type for variable a
type(a)

float

In [32]:
# Data type for variable b
type(b)

str

In [33]:
# Data type for variable c
type(c)

bool

# **Exercise 5: Python as arithmetic operator**

In [ ]:
# Given the following data
savings = 100
growth_multiplier = 1.1

In [34]:
# Assign product of savings and growth_multiplier to year1
year1=savings*growth_multiplier

In [36]:
# print out type of year1
print(type(year1))

<class 'float'>


In [37]:
# Assign sum of desc and desc to doubledesc
# Given desc = "compound interest"
doubledesc=desc+desc

In [39]:
# Print doubledesc
doubledesc

'compound interestcompound interest'

# **Exercise 6: Debugging the code**
The following code has error message.
Please fix it!


In [40]:
# Definition of savings and result
savings = 100
result = 100 * 1.10 ** 7

# Print out the message
print("I started with $" + (savings) + " and now have $" + result + ". Awesome!")

# Definition of pi_string
pi_string = "3.1415926"

# Convert pi_string into float: pi_float
float(pi_string)

TypeError: ignored

#### **Fixing in ACTION!!!**

In [41]:
# Definition of savings and result
result = 100 * 1.10 ** 7

# Fix the print out
print("I started with $" + str(savings) + " and now have $" + str(result) + ". Awesome!")

# Definition of pi_string
pi_string = "3.1415926"

# Convert pi_string into float: pi_float
pi_float = float(pi_string)

I started with $100 and now have $194.87171000000012. Awesome!


# **Which one of these will throw an error?**

In [42]:
# Summation of string
"I can add integers, like " + str(5) + " to strings."

'I can add integers, like 5 to strings.'

In [43]:
# Multiplication of string
"I said " + ("Hey " * 2) + "Hey!"

'I said Hey Hey Hey!'

In [44]:
# This one has an error
"The correct answer to this multiple choice exercise is answer number
" + 2

SyntaxError: ignored

In [46]:
# We can fix it
"The correct answer to this multiple choice exercise is answer number " + str(2)

'The correct answer to this multiple choice exercise is answer number 2'

In [47]:
# True = 1 and False = 0
True + False

1

# **Python Lists**
A list in a square bracket, [ ]

In [ ]:
# Different data type in a list


list

In [ ]:
# A list in a list


list

# **Exercise 7: Creating a list**

In [ ]:
# Area variables (in square meters)

# Create list areas


# Print areas


[11.25, 18.0, 20.0, 10.75, 9.5]


# **Exercise 8: Adding name to a list**

In [ ]:
# Area variables (in square meters)
hall = 11.25
kit = 18.0
liv = 20.0
bed = 10.75
bath = 9.50

# Adapt list areas - add name for each room


# Print areas


['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0, 'bedroom', 10.75, 'bathroom', 9.5]


# **Select the valid list**

In [ ]:
# Example A


[1, 3, 4, 2]

In [ ]:
# Example B


[[1, 2, 3], [4, 5, 7]]

In [ ]:
# Example C


[3, 'aaaaa', 3]

# **Exercise 9: List**

In [ ]:
# Area variables (in square meters)
hall = 11.25
kit = 18.0
liv = 20.0
bed = 10.75
bath = 9.50

# house information as list of lists
house = [["hallway", hall],
         ["kitchen", kit],
         ["living room", liv],
         ["bedroom", bed],
         ["bathroom", bath]]

# Print out house


# Print out the type of house


[['hallway', 11.25], ['kitchen', 18.0], ['living room', 20.0], ['bedroom', 10.75], ['bathroom', 9.5]]
<class 'list'>


# **Subsetting the lists**

In [ ]:
# Given the fam list from previous example
fam

['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89]

In [ ]:
# Emma's height


1.68

In [ ]:
# Using negative index to get Emma's height


1.68

# **List slicing**

In [ ]:
# Given the fam list again
fam

['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89]

In [ ]:
# What would fam[3:5] give us?


[1.68, 'mom']

In [ ]:
# What would fam[1:4] give us?


[1.73, 'emma', 1.68]

In [ ]:
# What would fam[:4] give us?


['liz', 1.73, 'emma', 1.68]

In [ ]:
# What would fam[5: ] give us?


[1.71, 'dad', 1.89]

# **Exercise 10 : Indexing**

In [ ]:
# Given the areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'living room',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 9.5]

In [ ]:
# Print out second element from areas


11.25


In [ ]:
# Print out last element from areas


9.5


In [ ]:
# Print out the area of the living room


20.0


# **Exercise 11: combination of list subsetting and variable assignment**

In [ ]:
# Given the areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'living room',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 9.5]

In [ ]:
# Sum of kitchen and bedroom area: eat_sleep_area


# Print the variable eat_sleep_area


28.75


# **Exercise 12: Use slicing to create a list**

In [ ]:
# Given the areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'living room',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 9.5]

In [ ]:
# Use slicing to create downstairs: first 6 elements of areas


['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0]

In [ ]:
# Use slicing to create upstairs: last 4 elements of areas


['bedroom', 10.75, 'bathroom', 9.5]

In [ ]:
# Print out downstairs and upstairs


['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0] ['bedroom', 10.75, 'bathroom', 9.5]


# **Exercise 13: Indexing again**

In [ ]:
# Given the areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'living room',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 9.5]

In [ ]:
# Alternative slicing to create downstairs: omitting the beginning index


['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0]

In [ ]:
# Alternative slicing to create upstairs: omitting the end index


['bedroom', 10.75, 'bathroom', 9.5]

In [ ]:
# Negative indexing for the last five elements


[20.0, 'bedroom', 10.75, 'bathroom', 9.5]

# **Subsetting lists of lists**

In [ ]:
# Given the following list
x = [["a", "b", "c"],
     ["d", "e","f"],
     ["g", "h", "i"]]

In [ ]:
# Example 1


'g'

In [ ]:
# Example 2


['g', 'h']

In [ ]:
# Given the house variable from previous example
house

[['hallway', 11.25],
 ['kitchen', 18.0],
 ['living room', 20.0],
 ['bedroom', 10.75],
 ['bathroom', 9.5]]

In [ ]:
# subsetting house[-1][1]


9.5

In [ ]:
# Data type


float

# **Manipulating Lists**

In [ ]:
# Given the fam list
fam

['liz', 1.73, 1.68, 'mom', 1.71, 'dad', 1.89]

In [ ]:
# Change of dad's height


['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.86]

In [ ]:
# Change of liz details


['lisa', 1.74, 'emma', 1.68, 'mom', 1.71, 'dad', 1.86]

# **Adding and removing elements in lists**


In [ ]:
# Given the items list
# We want to add additional ietms
items = ['cake', 'cookie', 'bread']
items + ['biscuit', 'tart']

['cake', 'cookie', 'bread', 'biscuit', 'tart']

In [ ]:
# However, the items are not updates


['cake', 'cookie', 'bread']

In [ ]:
# Create a new list to store the newly added items


['cake', 'cookie', 'bread', 'biscuit', 'tart']


In [ ]:
# Given the original fam list
fam = ["liz", 1.73, "emma", 1.68, "mom", 1.71, "dad", 1.89]
print(fam)

['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89]


In [ ]:
# A simple + sign would not add the data into the fam list permanently


['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89]

In [ ]:
# This would add the data into the list by assignning a new variable


['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89, 'me', 1.79]

In [ ]:
# Delete third element from fam list


In [ ]:
# Latest fam list - emma being removed


['liz', 1.73, 1.68, 'mom', 1.71, 'dad', 1.89]

#### We can use the **<font color='red'>plus-equals operator += </font>** to provide a convenient way to add a value **<font color='red'>to an existing variable</font>** and assign the new value **<font color='red'>back to the same variable</font>**.

In [ ]:
# Given the original fam list
fam = ["liz", 1.73, "emma", 1.68, "mom", 1.71, "dad", 1.89]

# Use of the plus-equals operator +=


['liz', 1.73, 'emma', 1.68, 'mom', 1.71, 'dad', 1.89, 'me', 1.7]


# **Exercise 14: Updating the list**

In [ ]:
# Given the areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'living room',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 9.5]

In [ ]:
# Update the bathroom area


In [ ]:
# Change "living room" to "chill zone"


In [ ]:
# Check the latest areas list
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'chill zone',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 10.5]

# **Exercise 15: Another example**

In [ ]:
# Given the original areas
areas = ["hallway", 11.25, "kitchen", 18.0, "chill zone", 20.0,
         "bedroom", 10.75, "bathroom", 10.50]
print(areas)

['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5]


In [ ]:
# Add poolhouse data to areas, new list is areas_1


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5]


In [ ]:
# Add garage data to areas_1, new list is areas_2


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5, 'garage', 15.45]


# **Quiz**

In [ ]:
# Given the areas
areas = areas_2[:]
print(areas)

# Output 1


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5, 'garage', 15.45]
['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 24.5, 15.45]


In [ ]:
# Given the areas
areas = areas_2[:]
print(areas)

# Output 2


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5, 'garage', 15.45]
['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 24.5, 'garage', 15.45]


In [ ]:
# Given the areas
areas = areas_2[:]
print(areas)

# Output 3


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5, 'garage', 15.45]
['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'garage', 15.45]


In [ ]:
# Given the areas
areas = areas_2[:]
print(areas)

# Output 4


['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 10.5, 'poolhouse', 24.5, 'garage', 15.45]
['hallway', 11.25, 'kitchen', 18.0, 'chill zone', 20.0, 'bedroom', 10.75, 'bathroom', 'poolhouse', 'garage', 15.45]


# **Your challenge**

In [ ]:
# Create list areas
areas = [11.25, 18.0, 20.0, 10.75, 9.50]

In [ ]:
# Create a copy by simply using equal sign


# change the second element of areas_copy to 21.1


# print the areas_and areas_copy


[11.25, 21.1, 20.0, 10.75, 9.5] [11.25, 21.1, 20.0, 10.75, 9.5]


In [ ]:
# To preserve the original areas data and make changes to areas_copy only


# Now make the changes to second element in areas_copy to 40


# Print both areas and areas_copy


[11.25, 21.1, 20.0, 10.75, 9.5] [11.25, 40, 20.0, 10.75, 9.5]


# **Good Luck and see you all next week!</p>**